core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -525,6 +525,15 @@ default Optional<ConnectorTableLayout> getNewTableLayout(ConnectorSession sessio
         return Optional.empty();
     }
 
+    /**
+     * Coerces the input type to the effective Trino type supported by the connector
+     * Return the effective {@link io.trino.spi.type.Type} that is supported by the connector for the given type, otherwise return {@link Optional#empty()}.
```

Comment:
`@return ...`

What does "otherwise return Optional.empty()" imply? That the type is not supported? Or that the connector supports exactly that type? Please spell that out clearly in the docs.